keras_nlp/layers/position_embedding.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Position embedding implementation based on `keras.layers.Layer`."""
+
+import tensorflow as tf
+from tensorflow import keras
+
+
+class PositionEmbedding(keras.layers.Layer):
+    """Creates a layer which learns a position embedding for inputs sequences.
+
+    This class assumes that in the input tensor, the last dimension corresponds
+    to the features, and the dimension before the last corresponds to the
+    sequence.
+
+    Args:
+      max_length: The maximum length of the dynamic sequence.
+      initializer: The initializer to use for the embedding weights. Defaults to
+        "glorot_uniform".
+      seq_axis: The axis of the input tensor where we add the embeddings.
+
+    Example:
+    ```python
+    class PositionalEmbedding(layers.Layer):
+        def __init__(self, max_length, vocab_size, embed_dim, **kwargs):
+            super().__init__(**kwargs)
+            self.token_embeddings = layers.Embedding(
+                input_dim=vocab_size, output_dim=embed_dim
+            )
+            self.position_embeddings = keras_nlp.layers.PositionEmbedding(
+                max_length=max_length
+            )
+
+        def call(self, inputs):
+            embedded_tokens = self.token_embeddings(inputs)
+            embedded_positions = self.position_embeddings(inputs)
```

Comment:
Shouldn't this be `self.position_embeddings(embedded_tokens)`?

Reply:
Yes, my bad, will fix